corgidrp/l1_to_l2a.py

```diff
@@ -282,22 +282,24 @@ def detect_cosmic_rays(input_dataset, detector_params, k_gain = None, sat_thresh
 
     return crmasked_dataset
 
-def correct_nonlinearity(input_dataset, non_lin_correction):
+def correct_nonlinearity(input_dataset, non_lin_correction, threshold=np.inf):
     """
-    Perform non-linearity correction of a dataset using the corresponding non-linearity correction
+    Perform non-linearity correction of a dataset using the corresponding non-linearity correction. We check for non-linear pixel and flag them in the DQ. 
 
     Args:
-        input_dataset (corgidrp.data.Dataset): a dataset of Images that need non-linearity correction (L2a-level)
-        non_lin_correction (corgidrp.data.NonLinearityCorrection): a NonLinearityCorrection calibration file to model the non-linearity
-
+        input_dataset (corgidrp.data.Dataset): a dataset of Images that need non-linearity correction (L2a-level).
+        non_lin_correction (corgidrp.data.NonLinearityCorrection): a NonLinearityCorrection calibration file to model the non-linearity.
+        threshold (float): threshold for flagging pixels in the DQ array, value above this threshold will be flagged in the DQ map as too nonlinear. By default it is set to infinity, user can change it to a different value.
+    
     Returns:
-        corgidrp.data.Dataset: a non-linearity corrected version of the input dataset
+        linearized_dataset (corgidrp.data.Dataset): A non-linearity corrected version of the input dataset
```

Comment:
oops I missed this: this actually isn't the Google-style syntax that we use. The Returns value is not named. (i.e. remove linearized_dataset)